sys/kern/src/arch/rv64/trap.rs

```diff
@@ -198,13 +198,12 @@ fn trap_handler(task: &mut task::Task) {
             unsafe {
                 // Advance program counter past ecall instruction.
                 let epc = register::mepc::read() as u64 + 4;
```

Comment:
"+4" this is assuming no compressed instructions. Something to track for future and any other manipulations of pc.

Reply:
I think `mepc` is at an `ecall` at this point, so I don't think there's any issue in this specific case.

Reply:
> "+4" this is assuming no compressed instructions. Something to track for future and any other manipulations of pc.

That's right. As Fawaz pointed out this would be just for the ECALL path, so `+4` works there. I will add a comment to capture this.

Reply:
Pushed a change capturing this comment.